src/index/ivf_raft/ivf_raft.cuh

```diff
@@ -344,12 +373,150 @@ class RaftIvfIndexNode : public IndexNode {
 
     virtual Status
     Serialize(BinarySet& binset) const override {
-        return Status::not_implemented;
+        if (!gpu_index_.has_value())
+            return Status::empty_index;
+        std::stringbuf buf;
+
+        std::ostream os(&buf);
+
+        os.write((char*)(&this->dim_), sizeof(this->dim_));
```

Comment:
This whole serialization setup seems fine, but I wonder if we wouldn't make our lives easier by upgrading to RAFT 23.04 and just invoking the [serialization API](https://github.com/rapidsai/raft/pull/1315) directly from that. It would likely dramatically simplify this logic. That can also happen as a follow up, if this is working for now.

Reply:
It would also centralize this so that, for example, you could load an index into Milvus that's been previously trained w/ RAFT.